addons/event-audit/common/src/main/java/org/commonjava/indy/event/audit/EventAuditListener.java

```diff
@@ -0,0 +1,227 @@
+        try
+        {
+            StoreKey affectedStore = keyedLocation.getKey();
+            final Transfer txfr = downloadManager.getStorageReference( keyedLocation.getKey(), transfer.getPath() );
+            if ( txfr != null )
+            {
+
+                if ( StoreType.remote == affectedStore.getType() )
+                {
+                    final RemoteRepository repo = (RemoteRepository) storeManager.getArtifactStore( affectedStore );
+                    if ( repo != null )
+                    {
+                        fileEvent.setTargetLocation( repo.getUrl() );
+                    }
+                }
+
+                TransferMetadata artifactData = contentDigester.digest( affectedStore, path, event.getEventMetadata() );
+                Map<String, String> extra = new HashMap<>();
+                extra.put( ContentDigest.MD5.name(), artifactData.getDigests().get( ContentDigest.MD5 ) );
+                extra.put( ContentDigest.SHA_1.name(), artifactData.getDigests().get( ContentDigest.SHA_1 ) );
+                extra.put( ContentDigest.SHA_256.name(), artifactData.getDigests().get( ContentDigest.SHA_256 ) );
+                extra.put( "Size", String.valueOf( artifactData.getSize() ) );
+
+                fileEvent.setChecksum( artifactData.getDigests().get( ContentDigest.MD5 ) );
+
+                fileEvent.setExtra( extra );
+            }
+
+        }
+        catch ( final IndyWorkflowException | IndyDataException e )
+        {
+            logger.error( String.format( "Failed to transform file event. Reason: %s", e.getMessage() ), e );
+        }
+    }
+
+    private void transformFileGroupingEvent( PromoteCompleteEvent event, FileGroupingEvent fileGroupingEvent )
+    {
+
+        PathsPromoteCompleteEvent pathsPromoteCompleteEvent = null;
+        if ( event instanceof PathsPromoteCompleteEvent )
+        {
+            pathsPromoteCompleteEvent = (PathsPromoteCompleteEvent) event;
+        }
+
+        PathsPromoteResult promoteResult = pathsPromoteCompleteEvent.getPromoteResult();
+
+        String error = promoteResult.getError();
+        if ( error != null )
+        {
+            logger.trace( "Error in promoteResult, skip audit." );
+            return;
+        }
+
+        Set<String> paths = promoteResult.getCompletedPaths();
+        if ( paths.isEmpty() )
+        {
+            logger.trace( "No completedPaths, skip audit." );
+            return;
+        }
+
+        PathsPromoteRequest req = promoteResult.getRequest();
+        StoreKey source = req.getSource();
+        StoreKey target = req.getTarget();
+
+        Map<String, String> extra = new HashMap<>();
+        extra.put( "source", source.toString() );
+        extra.put( "target", target.toString() );
+
+        fileGroupingEvent.setExtra( extra );
+        fileGroupingEvent.setTimestamp( new Date() );
+
+        TrackingKey trackingKey = getTrackingKey( source );
+        if ( trackingKey == null )
```

Comment:
Just want to double confirm for this condition. Should this be put at the top of this method? Or it should be here?